tests/testthat/test-try_stack.R

```diff
@@ -25,6 +25,16 @@ describe("with_retries", {
       "not enough retries"
     )
   })
-
-
 })
+
+describe("try_with_exit_code", {
+    test_that("it evaluates expr successfully", {
+    expect_true(try_with_exit_code(sum(1, 2)))
+  })
+    test_that("Expr evaluation fails and it exits R with specific exit code", {
+    with_mock(
+      `q` = function(status, ...) stop(status),
```

Comment:
`warning` shouldn't be necessary... all the relevant information should be in the `exit_code`